PocketKit/Sources/PocketKit/Settings/AccountViewModel.swift

```diff
@@ -18,8 +23,22 @@ class AccountViewModel: ObservableObject {
         appSession.currentSession = nil
     }
 
+    func toggleAppBadge() {
+        UNUserNotificationCenter.current().requestAuthorization(options: .badge) {
+            (granted, error) in
+            if error == nil && granted == true {
+                let currentValue = self.userDefaults.bool(forKey: AccountViewModel.ToggleAppBadgeKey)
+                self.userDefaults.setValue(!currentValue, forKey: AccountViewModel.ToggleAppBadgeKey)
+                self.notificationCenter.post(name: .listUpdated, object: nil)
+            } else {
+                self.userDefaults.set(false, forKey: "SettingsRowToggle")
+            }
```

Comment:
style preference, but using `guard` makes it easier for me to read

```
 guard error == nil && granted == true else {
                self.userDefaults.set(false, forKey: "SettingsRowToggle")
                return
            }

            let currentValue = self.userDefaults.bool(forKey: AccountViewModel.ToggleAppBadgeKey)
            self.userDefaults.setValue(!currentValue, forKey: AccountViewModel.ToggleAppBadgeKey)
            self.notificationCenter.post(name: .listUpdated, object: nil)
 ```
 
Now that I think about it, maybe the post here doesn't make as much sense as we originally thought since the list hasn't been updated, so maybe we can change the name or observe that the userDefaults changed its value. Just a thought to consider not a blocker for me